#### Mục tiêu
Thực hành lọc dữ liệu theo điều kiện và thực hiện các công thức toán học theo nhóm. 

Bộ dữ liệu House Price – Đống Đa chứa thông tin mua bán nhà quận Đống đa, các thông tin được thống kê bao gồm:

Title: Tiêu đề  

Address: địa chỉ

Area: khu vực

Price: giá

postDate: ngày đăng

land_certificate: giấy chứng nhận

house_direction: hướng nhà

balcony_direction: hướng ban công

toilet: số phòng vệ sinh

bedroom: số phòng ngủ

floor: số tầng

type_of_land: loại nhà đất

street_name: tên đường

ward_name: tên phường

district_name: tên quận

city_name: tên thành phố

lat: vĩ độ

long: kinh độ

### Yêu cầu
Đọc bộ dữ liệu, cho biết số dòng, số cột và kiểu dữ liệu của các thuộc tính.


In [1]:
# Nhâp thư viện
import pandas as pd
import seaborn as sns
import numpy as np

In [3]:
# Đọc dữ liệu

df = pd.read_csv('house_price_Dống-Da_Hà-Nội_subdata.csv')
df.head()

area    price  postDate land_certificate house_direction balcony_direction  \
0  50.0  14700.0     44197            So do             NaN               NaN   
1  57.0   4200.0     44265            So do             Nam               Nam   
2  62.0  11000.0     44268            So do             NaN          dong-Nam   
3  26.0   2900.0     44292            So do             NaN               NaN   
4  40.0   3950.0     44292            So do             NaN               NaN   

   toilet  bedroom  floor        lat        long property_type  
0     4.0      6.0    6.0  21.015698  105.824583     trong ngo  
1     3.0      8.0    4.0  21.025208  105.833111     trong ngo  
2     6.0      5.0    6.0  21.025379  105.801011     trong ngo  
3     4.0      3.0    4.0  21.014325  105.834918     trong ngo  
4     NaN      NaN    5.0  21.009862  105.838083     trong ngo

In [4]:
# số dòng và cột của dữ liệu
df.shape

(671, 12)

In [5]:
# xem kiểu dữ liệu của các cột

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 671 entries, 0 to 670
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   area               665 non-null    float64
 1   price              617 non-null    float64
 2   postDate           671 non-null    int64  
 3   land_certificate   369 non-null    object 
 4   house_direction    42 non-null     object 
 5   balcony_direction  24 non-null     object 
 6   toilet             240 non-null    float64
 7   bedroom            320 non-null    float64
 8   floor              366 non-null    float64
 9   lat                671 non-null    float64
 10  long               671 non-null    float64
 11  property_type      671 non-null    object 
dtypes: float64(7), int64(1), object(4)
memory usage: 63.0+ KB


#### Nhận xét:

Dữ liệu có 671 dòng, 12 cột

Những cột postDate, lat, long, property_type: dữ liệu đủ

còn lại những cột khác đều thiếu dữ liệu

### Lọc ra các bản ghi bán nhà riêng tại phường Trung liệt hoặc phường Khâm Thiên


### Lọc các thông tin Địa chỉ, Giá, Hướng nhà, Hướng ban công của các bản ghi có giấy chứng nhận sổ đỏ và có 3 phòng ngủ trở lên.


In [6]:
df1 = df[(df['land_certificate'] == 'So do') & (df['bedroom'] > 3)]
df1[['price', 'house_direction', 'balcony_direction', 'land_certificate', 'bedroom']]

price house_direction balcony_direction land_certificate  bedroom
0    14700.0             NaN               NaN            So do      6.0
1     4200.0             Nam               Nam            So do      8.0
2    11000.0             NaN          dong-Nam            So do      5.0
6     5280.0         Tay-Nam               NaN            So do      4.0
15    3800.0             NaN               NaN            So do      4.0
..       ...             ...               ...              ...      ...
646  30600.0             NaN               NaN            So do      6.0
653  30000.0        dong-Bac               NaN            So do      5.0
659  12000.0        dong-Nam               NaN            So do      4.0
663  72000.0             NaN               NaN            So do      6.0
667  17500.0             NaN               NaN            So do      6.0

[143 rows x 5 columns]

### Với mỗi loại nhà đất, tính trung bình cộng giá cũng như giá lớn nhất và giá nhỏ nhất.


In [7]:
df2 = df.groupby('property_type')['price'].agg(['mean', 'max', 'min'])
df2

mean        max    min
property_type                                
mat pho        69209.237619  3300000.0    4.5
trong ngo       8143.431450   120000.0  450.0

### Tính trung bình cộng số phòng ngủ, số phòng vệ sinh, số tầng của mỗi phường.